### ISSUE TITLE
Multi expression "try" form discards some of them

### ISSUE THREAD

**Issue by etanol** [hylang/hy#1584]
When using multiple expressions in the body of a `try` form, those that are function calls seem to be discarded by the compiler.  Given the following code sample:

``` hy
(setv val 0)
(print (try
         (+= val 1)
         (print "Check")
         val
         (except [Exception] -1)))
```

Executing with Hy 0.14.0 under Python 3.5.2 prints `1` but **not** `Check`.  Running `hy2py3` prints the following code:

``` python
val = 0
try:
    val += 1
    _hy_anon_var_1 = val
except Exception:
    _hy_anon_var_1 = -1
print(_hy_anon_var_1)
```

Using a `do` form solves the issue.  However, the documentation seems to suggest that multiple expressions are allowed in the `try` body; as well as #1427.

**Comment by Kodiologist** [comment]
Yep, this is a bug and it's in master. (Remember to check master, not the last stable release, for bugs before reporting.) It looks like statements are kept (`((try (raise 1) (raise 2) (finally 1))`) but expressions are dropped (`(try x (raise 2) (finally 1))`).

**Comment by gilch** [comment]
Probably related to #1542. Whack-a-mole.
